File: src/lib/api/brokerProxyClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class BrokerProxy {
  constructor(private readonly host: RPCHost) {}

  public open(
    arg: pb.IBrokerProxyRequest = new pb.BrokerProxyRequest()
  ): GenericReadable<pb.BrokerProxyEvent> {
    return this.host.expectMany(this.host.call("BrokerProxy/Open", new pb.BrokerProxyRequest(arg)));
  }
  public sendKeys(
    arg: pb.IBrokerProxySendKeysRequest = new pb.BrokerProxySendKeysRequest()
  ): Promise<pb.BrokerProxySendKeysResponse> {
    return this.host.expectOne(
      this.host.call("BrokerProxy/SendKeys", new pb.BrokerProxySendKeysRequest(arg))
    );
  }
  public receiveKeys(
    arg: pb.IBrokerProxyReceiveKeysRequest = new pb.BrokerProxyReceiveKeysRequest()
  ): Promise<pb.BrokerProxyReceiveKeysResponse> {
    return this.host.expectOne(
      this.host.call("BrokerProxy/ReceiveKeys", new pb.BrokerProxyReceiveKeysRequest(arg))
    );
  }
  public data(
    arg: pb.IBrokerProxyDataRequest = new pb.BrokerProxyDataRequest()
  ): Promise<pb.BrokerProxyDataResponse> {
    return this.host.expectOne(
      this.host.call("BrokerProxy/Data", new pb.BrokerProxyDataRequest(arg))
    );
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/BrokerProxyClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class BrokerProxyClient(filepath: String) : RPCClient(filepath) {

    suspend fun open(
        arg: BrokerProxyRequest = BrokerProxyRequest()
    ): RPCResponseStream<BrokerProxyEvent> =
        this.callStreaming("BrokerProxy/Open", arg)

    suspend fun sendKeys(
        arg: BrokerProxySendKeysRequest = BrokerProxySendKeysRequest()
    ): BrokerProxySendKeysResponse =
        this.callUnary("BrokerProxy/SendKeys", arg)

    suspend fun receiveKeys(
        arg: BrokerProxyReceiveKeysRequest = BrokerProxyReceiveKeysRequest()
    ): BrokerProxyReceiveKeysResponse =
        this.callUnary("BrokerProxy/ReceiveKeys", arg)

    suspend fun data(
        arg: BrokerProxyDataRequest = BrokerProxyDataRequest()
    ): BrokerProxyDataResponse =
        this.callUnary("BrokerProxy/Data", arg)

    suspend fun close(
        arg: BrokerProxyCloseRequest = BrokerProxyCloseRequest()
    ): BrokerProxyCloseResponse =
        this.callUnary("BrokerProxy/Close", arg)

}
